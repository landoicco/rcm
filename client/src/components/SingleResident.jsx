const SingleResident = ({ item }) => {
  return (
    <div>
      <div className="card blue-grey darken-1">
        <div className="card-content white-text">
          <span className="card-title">
            {item.firstName} {item.lastName}
          </span>
        </div>
        <div className="card-action">
          <blockquote>{item.address}</blockquote>
          <p>Phone number: {item.phoneNumber}</p>
          <p>Email: {item.email}</p>
        </div>
      </div>
    </div>
  );
};

export default SingleResident;
