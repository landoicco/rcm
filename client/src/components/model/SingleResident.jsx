//@flow

import React from "react";
import type { ResidentType } from "../../commons/types";

const SingleResident = ({ item }: ResidentType): React.Node => {
  return (
    <div>
      <div className="card teal lighten-4">
        <div className="card-content">
          <span className="card-title">
            {item.firstName} {item.lastName}
          </span>
        </div>
        <div className="card-action">
          <p>Address: {item.address}</p>
          <p>Phone number: {item.phoneNumber}</p>
          <p>Email: {item.email}</p>
        </div>
      </div>
    </div>
  );
};

export default SingleResident;
