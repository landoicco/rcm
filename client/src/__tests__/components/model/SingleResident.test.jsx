import { render, screen } from "@testing-library/react";
import SingleResident from "../../../components/model/SingleResident";

const mockResident = {
  id: "-1",
  firstName: "Test",
  lastName: "Tester",
  address: "here",
  residence: "here 000",
  email: "test@mail.com",
  phoneNumber: "345345",
};

describe("SingleResident", () => {
  test("The component render correctly", () => {
    const { container } = render(<SingleResident item={mockResident} />);
    expect(container).toMatchSnapshot();
  });
});
