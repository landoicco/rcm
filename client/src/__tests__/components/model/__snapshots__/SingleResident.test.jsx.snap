// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleResident The component render correctly 1`] = `
<div>
  <div>
    <div
      class="card teal lighten-4"
    >
      <div
        class="card-content"
      >
        <span
          class="card-title"
        >
          Test
           
          Tester
        </span>
      </div>
      <div
        class="card-action"
      >
        <p>
          Address: 
          here
        </p>
        <p>
          Phone number: 
          345345
        </p>
        <p>
          Email: 
          test@mail.com
        </p>
      </div>
    </div>
  </div>
</div>
`;
